int MAIN_ENTRY(CPPExceptionHandling);